src/main/scala/mesosphere/marathon/core/launchqueue/impl/OfferConstraintsStreamLogic.scala

```diff
@@ -27,15 +29,6 @@ object OfferConstraintsStreamLogic extends StrictLogging {
       .buffer(1, OverflowStrategy.dropHead) // While we are back-pressured, we drop older interim frames
       .via(RateLimiterFlow.apply(minUpdateInterval))
       .map(_.roleState)
-      .sliding(2)
```

Comment:
@timcharper 
The intent behind these lines was removing redundant constraints "updates" that do not change constraints.
Without that, an instance state change that doesn't result in constraints change might nevertheless cause reissuing the latest UpdateFramework with the same offer constraints.
This is exactly what broke integration tests in the last run: https://jenkins.mesosphere.com/service/jenkins/job/marathon-pipelines/job/PR-7229/15/console

Do you see a more transparent way to accomplish the same thing? 
If no, then it looks like I'll have to add a good comment here.